linode_api4/objects/linode.py

```diff
@@ -765,20 +790,75 @@ def kvmify(self):
 
         return True
 
-    def mutate(self):
+    def mutate(self, allow_auto_disk_resize=True):
         """
         Upgrades this Instance to the latest generation type
         """
-        self._client.post('{}/mutate'.format(Instance.api_endpoint), model=self)
+
+        params = {
+            "allow_auto_disk_resize": allow_auto_disk_resize
+        }
+
+        self._client.post('{}/mutate'.format(Instance.api_endpoint), model=self, data=params)
 
         return True
 
-    def initiate_migration(self):
+    def initiate_migration(self, region=None, upgrade=None):
         """
         Initiates a pending migration that is already scheduled for this Linode
         Instance
         """
-        self._client.post('{}/migrate'.format(Instance.api_endpoint), model=self)
+        params = {
+            "region": region.id if issubclass(type(region), Base) else region,
+            "upgrade": upgrade
+        }
+
+        util.drop_null_keys(params)
+
+        self._client.post('{}/migrate'.format(Instance.api_endpoint), model=self, data=params)
+
+    def firewalls(self):
+        """
+        View Firewall information for Firewalls associated with this Linode.
+        """
+        from linode_api4.objects import Firewall # pylint: disable=import-outside-toplevel
+
+        result = self._client.get('{}/firewalls'.format(Instance.api_endpoint), model=self)
+
+        firewalls = []
+        for firewall in result:
+            firewalls.append(Firewall(self._client, firewall["id"]))
```

Comment:
Since this endpoint uses the [list return structure](https://www.linode.com/docs/api/linode-instances/#firewalls-list__response-samplesl), we'll need to loop through `result["data"]` rather than `result`.

We can also make use of list comprehensions here, although that's more of a stylistic/optional change:

```python
firewalls = [Firewall(self._client, firewall["id"]) for firewall in result["data"]]
```